src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -0,0 +1,197 @@
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.update;
+
+public class TOTPQueries {
+    public static String getQueryToCreateUserDevicesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUserDevicesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL," + "device_name VARCHAR(256) NOT NULL,"
+                + "secret_key VARCHAR(256) NOT NULL,"
+                + "period INTEGER NOT NULL," + "skew INTEGER NOT NULL," + "verified BOOLEAN NOT NULL,"
+                + "PRIMARY KEY (user_id, device_name))";
+    }
+
+    public static String getQueryToCreateUsedCodesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUsedCodesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL, " + "device_name VARCHAR(256), "
+                + "code CHAR(6) NOT NULL," + "is_valid_code BOOLEAN NOT NULL,"
+                + "expiry_time_ms BIGINT UNSIGNED NOT NULL," // Note: UNSIGNED won't work in Postgres
+                + "FOREIGN KEY (user_id, device_name) REFERENCES " + Config.getConfig(start).getTotpUserDevicesTable()
+                + "(user_id, device_name) ON DELETE CASCADE);";
+    }
+
+    public static String getQueryToCreateUsedCodesIndex(Start start) {
+        return "CREATE INDEX IF NOT EXISTS totp_used_codes_expiry_time_ms_index ON "
+                + Config.getConfig(start).getTotpUsedCodesTable() + " (expiry_time_ms)";
+    }
+
+    public static void createDevice(Start start, TOTPDevice device)
+            throws StorageQueryException, SQLException {
+        String QUERY = "INSERT INTO " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " (user_id, device_name, secret_key, period, skew, verified) VALUES (?, ?, ?, ?, ?, ?)";
+
+        update(start, QUERY, pst -> {
+            pst.setString(1, device.userId);
+            pst.setString(2, device.deviceName);
+            pst.setString(3, device.secretKey);
+            pst.setInt(4, device.period);
+            pst.setInt(5, device.skew);
+            pst.setBoolean(6, device.verified);
+        });
+    }
+
+    public static int markDeviceAsVerified(Start start, String userId, String deviceName)
+            throws StorageQueryException, SQLException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " SET verified = true WHERE user_id = ? AND device_name = ? WHERE verified = false;";
+        return update(start, QUERY, pst -> {
+            pst.setString(1, userId);
+            pst.setString(2, deviceName);
+        });
+    }
+
+    public static int deleteDevice(Start start, String userId, String deviceName)
+            throws StorageQueryException, SQLException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ? AND device_name = ?;";
+
```

Comment:
Need to move logic of removeUsedCodes in here (in a transaction)

Reply:
Done.